### PR TITLE
Try to make the cron help clearer

### PR DESCRIPTION
**Pull request by jsoref**
The rendering of https://www.jenkins.io/doc/book/pipeline/syntax/#triggers without these changes was too easy for me to confuse while blurry-eyed.

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@daniel-beck 

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [x] There are at least 2 approvals for the pull request and no outstanding requests for change
- [x] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [x] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [x] Proper changelog labels are set so that the changelog can be generated automatically
- [x] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Review by jvz — APPROVED** [review]

File: core/src/main/resources/hudson/triggers/TimerTrigger/help-spec.jelly [hunk 1]
```diff
@@ -71,15 +71,15 @@
...
+# Every ten minutes in the first half of every hour (three times, perhaps at :04, :14, :24):
 H(0-29)/10 * * * *
-# once every two hours at 45 minutes past the hour starting at 9:45 AM and finishing at 3:45 PM every weekday.
+# Once every two hours at 45 minutes past the hour starting at 9:45 AM and finishing at 3:45 PM every weekday:
 45 9-16/2 * * 1-5
-# once in every two hours slot between 9 AM and 5 PM every weekday (perhaps at 10:38 AM, 12:38 PM, 2:38 PM, 4:38 PM)
-H H(9-16)/2 * * 1-5
-# once a day on the 1st and 15th of every month except December
+# Once in every two hour slot between 8 AM and 4 PM every weekday (perhaps at 9:38 AM, 11:38 AM, 1:38 PM, 3:38 PM):
+H H(8-15)/2 * * 1-5
```
Comment (daniel-beck) [inline core/src/main/resources/hudson/triggers/TimerTrigger/help-spec.jelly:81]: Are you sure this can ever be odd hours? Have you tried this?

Comment (jsoref) [reply core/src/main/resources/hudson/triggers/TimerTrigger/help-spec.jelly:81]: I haven't tried. But if it doesn't support this, then the description should be change to explain that you can't do this.

Normal crontab apparently doesn't distinguish between odd/even:
[“At minute 0 past every 2nd hour from 8 through 15.”](https://crontab.guru/#0_8-15/2_*_*_*)
[“At minute 0 past every 2nd hour from 9 through 16.”](https://crontab.guru/#0_9-16/2_*_*_*)

And note that one of the items talked about the problem of using `days/3`.

**Review by varyvol — APPROVED** [review]

File: core/src/main/resources/hudson/triggers/TimerTrigger/help-spec.jelly (hunk 1 above)
Comment (MRamonLeon) [reply core/src/main/resources/hudson/triggers/TimerTrigger/help-spec.jelly:81]: I don't get why the description says from 8 to 4 and the expression is from 8 to 3. The links you provided have a congruent description and expression, not like our code. I mean, the comment was wrong before and now it stays so. WDYT?

Comment (jsoref) [reply core/src/main/resources/hudson/triggers/TimerTrigger/help-spec.jelly:81]: I tried to reason through the comment and gave up. I had a theory but... I probably should have opened a ticket about it.

Comment (daniel-beck) [reply core/src/main/resources/hudson/triggers/TimerTrigger/help-spec.jelly:81]: @MRamonLeon See the examples provided (which are a possibility depending on how it's implemented, I'm just not confident it works as explained). Clearly 3:38 PM isn't before 3 PM.

Comment (daniel-beck) [reply core/src/main/resources/hudson/triggers/TimerTrigger/help-spec.jelly:81]: @oleg-nenashev What specifically is the defect?

Comment (oleg-nenashev) [reply core/src/main/resources/hudson/triggers/TimerTrigger/help-spec.jelly:81]: Sorry, I was referring to 

> Normal crontab apparently doesn't distinguish between odd/even:

IIRC we had issues with that in the past. Anyway, I do not block it in any means

**Review by oleg-nenashev — APPROVED** [review]

**Review by MRamonLeon — APPROVED** [review]

**Pull request merged**